src/csv/builder.cr

```diff
@@ -138,7 +139,7 @@ class CSV::Builder
     private def needs_quotes?(value)
       value.each_byte do |byte|
         case byte.unsafe_chr
-        when ',', '\n', '"'
+        when @separator, @quote_char, '\n'
           return true
         end
```

Comment:
A `case` condition is a bit overkill here.
Smaller to write `return true if [@separator, @quote_char, '\n'].includes? byte.unsafe_chr`

Reply:
... but slower

Reply:
Thanks @straight-shoota I didn't know that, noted! Why `case` is more perfomant, because  the includes use an `Array`?

Reply:
`Array#includes?` needs to setup and execute a loop and several function calls... not even considering the array allocation. While you could use a tuple instead and LLVM will probably optimize some of this, there is still a considerable overhead over a `case ... when` which simply translates to a few conditionals.